framework/src/userobjects/TimedSubdomainModifier.C

```diff
@@ -0,0 +1,319 @@
+  params.addParam<size_t>(
+      "blocks_from_column_index", 1, "Zero-based index of the blocks_from column. Default is '1'.");
+  params.addParam<size_t>(
+      "blocks_to_column_index", 2, "Zero-based index of the blocks_to column. Default is '2'.");
+  params.addParam<std::string>("time_column_text", "Header text of the time column.");
+  params.addParam<std::string>("blocks_from_column_text", "Header text of the blocks_from column.");
+  params.addParam<std::string>("blocks_to_column_text", "Header text of the blocks_to column.");
+  params.addParamNamesToGroup(
+      "data_file header delimiter comment time_column_index blocks_from_column_index "
+      "blocks_to_column_index time_column_text blocks_from_column_text blocks_to_column_text",
+      "Data input from file.");
+
+  params.addClassDescription("Moves all elements of a block to another at the given times.");
+
+  return params;
+}
+
+TimedSubdomainModifier::TimedSubdomainModifier(const InputParameters & parameters)
+  : TimedElementSubdomainModifier(parameters)
+{
+  // determine function arguments
+
+  int bFromData_File =
+      isParamValid("data_file") + isParamValid("header") + isParamValid("delimiter") +
+      isParamValid("comment") + isParamValid("time_column_index") +
+      isParamValid("blocks_from_column_index") + isParamValid("blocks_to_column_index") +
+      isParamValid("time_column_text") + isParamValid("blocks_from_column_text") +
+      isParamValid("blocks_to_column_text");
+
+  int bFromData_File_NeedsHeader = isParamValid("time_column_text") +
+                                   isParamValid("blocks_from_column_text") +
+                                   isParamValid("blocks_to_column_text");
+
+  int bFromParameters =
+      isParamValid("times") + isParamValid("blocks_from") + isParamValid("blocks_to");
+
+  // Check parameter set for inconsistencies
+  int nFrom = ((bFromData_File > 0) + (bFromParameters > 0));
+  if (nFrom != 1)
+  {
+    mooseError("Data on times and blocks must be provided either via a CSV file ('data_file' and "
+               "corresponding blocks), or via direct parameter input ('times', 'blocks_from', and "
+               "'blocks_to').");
+  };
+
+  if (bFromParameters > 0)
+  {
+    if (bFromParameters != 3)
+    {
+      mooseError("All parameters 'times', and 'blocks_from', and 'blocks_to' must be specified.");
+    };
+    buildFromParameters();
+  }
+  else if (bFromData_File > 0)
+  {
+    if ((isParamSetByUser("time_column_index") + isParamSetByUser("time_column_text")) > 1)
+    {
+      mooseError(
+          "The parameters 'time_column_index', and 'time_column_text' are mutual exclusive.");
+    };
+    if ((isParamSetByUser("blocks_from_column_index") +
+         isParamSetByUser("blocks_from_column_text")) > 1)
+    {
+      mooseError("The parameters 'blocks_from_column_index', and 'blocks_from_column_text' are "
+                 "mutual exclusive.");
+    };
```

Comment:
```suggestion
    }
```
(delete the whole bracket, but for future reference you dont need this semicolon